src/Download_V2.js

```diff
@@ -92,7 +92,7 @@ export default class Download_V2 extends React.PureComponent {
     };
     DeviceEventEmitter.addListener('dldProgress', progressListener);
     return {
-      remove: () => DeviceEventEmitter.removeListener('dldProgress', progressListener)
+      remove: () => DeviceEventEmitter.removeAllListeners('dldProgress', progressListener)
```

Comment:
@MeenuDhanoa did you test downloading more lessons one after another after changing the listener removing?